azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -759,14 +759,21 @@ func expandAzureRmKubernetesClusterAgentProfiles(d *schema.ResourceData) []conta
 	name := config["name"].(string)
 	count := int32(config["count"].(int))
 	vmSize := config["vm_size"].(string)
-	osDiskSizeGB := int32(config["os_disk_size_gb"].(int))
 	osType := config["os_type"].(string)
 
+	// If this value is not provided, the default value read from configuration will be 0.
+	// AKS API prefers to accept nil pointer instead of 0.
+	osDiskSizeGB := int32(config["os_disk_size_gb"].(int))
+	var osDiskSizeGBPtr *int32 = nil
+	if osDiskSizeGB > 0 {
```

Comment:
A simpler (and more common pattern in the provider) way of doing this is after the profile def:

```go
if osDiskSizeGB > 0 {
    profile.OsDiskSizeGB = osDiskSizeGB
}
```

as profile.OsDiskSizeGB will be initialized to `nil`

Reply:
Very good suggestion and I've updated it.